app/src/lib/stores/app-store.ts

```diff
@@ -1607,13 +1608,62 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
       return { workingDirectory, selectedFileIDs, diff }
     })
+
+    this._triggerMergeConflictsFlow(repository, status)
+
     this.emitUpdate()
 
     this.updateChangesDiffForCurrentSelection(repository)
 
     return true
   }
 
+  /** starts the conflict resolution flow, if appropriate */
+  private async _triggerMergeConflictsFlow(
+    repository: Repository,
+    status: IStatusResult
+  ) {
+    const inConflictedMerge = status.workingDirectory.files.some(f => {
+      return (
+        f.status === AppFileStatus.Conflicted ||
+        f.status === AppFileStatus.Resolved
+      )
+    })
+    if (!inConflictedMerge) {
+      return
+    }
+    if (status.currentBranch === undefined) {
+      return
+    }
+    // if we're already in the conflict merge resolution flow, bail
+    if (
+      this.currentPopup !== null &&
+      this.currentPopup.type !== PopupType.MergeConflicts &&
+      this.currentPopup.type !== PopupType.AbortMerge
+    ) {
+      return
+    }
+    try {
+      const possibleTheirsBranches = await getBranchesPointedAt(
+        repository,
+        'MERGE_HEAD'
+      )
+      const theirBranch =
+        possibleTheirsBranches.length === 1
+          ? possibleTheirsBranches[0]
+          : undefined
+      const ourBranch = status.currentBranch
+      this._showPopup({
+        type: PopupType.MergeConflicts,
+        repository,
+        ourBranch,
+        theirBranch,
+      })
+    } catch (e) {
+      log.info(e)
```

Comment:
Can you also provide a summary message here about the context alongside logging the error?

Or if we don't need it and can push the error handling into `getBranchesPointedAt` that will simplify things.